cirq/protocols/approximate_equality.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+import numpy as np
+
+from typing_extensions import Protocol
+
+
+class SupportsApproximateEquality(Protocol):
+    """Object which can be compared approximately."""
+
+    def _approx_eq_(
+            self,
+            other: Any,
+            *,
+            atol: float
+        ) -> bool:
+        """Approximate comparator.
+
+        Types implementing this protocol define their own logic for approximate
+        comparison with other types.
+
+        Args:
+            other: Target object for approximate comparison.
+            atol: The minimum absolute tolerance. See np.isclose() documentation
+                  for details.
+
+        Returns:
+            True if objects are approximately equal, False otherwise. Returns
+            NotImplemented when approximate equality is not implemented for
+            given types.
+        """
+        pass
+
+
+def approx_eq(
+        val: Any,
+        other: Any,
+        *,
+        atol: float = 0.0) -> bool:
```

Comment:
Don't specify a default value for `atol`. If a user wants exact equality they'll use `==`.

Reply:
Done.